file-attachments.mdx

```diff
@@ -0,0 +1,184 @@
+   alt: 'File Attachments: Databases can now store files and images'
+ author: Sorin Toma
+ date: 08-30-2023
+ published: true
+ slug: file-attachments
+ ogImage: https://raw.githubusercontent.com/xataio/mdx-blog/main/images/fa_blog_illustration.jpeg
+---
+
+Today, as part of [our launch week](https://xata.io/blog/launch-week-august-2023), we’re beyond excited to announce a feature that we've wanted to add ever since we started Xata: File Attachments. Think of it as having a new database column type where you can store files of any size, and behind the scenes they are stored in AWS S3 and cached through a global CDN. With this approach, files simply become part of a database record. For example, they respect the same security boundary -- if you delete a record, the files also get deleted. Image file types also get some extra functionality allowing you to request them at any size and style with built-in transformations.
+
+Whether you’re building a full scale e-commerce application, a personal blog or evolving your product to be AI-native — file attachments unlock a number of new use cases while providing an opportunity to simplify your architecture and reduce the number of services you rely on.
+
+In this blog we'll dive into the capabilities released today and architecture behind our implementation. We hope you enjoy the read as much as we did building it all out 😃
+
+### Adding file storage to the database
+
+Both files and relational data are ubiquitous in todays applications. Even the most basic scenarios generally require both structured relational data and file storage. Think about a blog which uses both posts metadata and images, a product catalogue or a document library. They all require basic data record management and query capabilities, as well as storage and access to large binary objects like images, videos, and documents.
+
+We often see engineers using a relational database with a separate storage service to store the data used by their application. In most cases the binary files are related to the relational data, therefore the common pattern is to store a file link in the database. This experience adds unnecessary friction and we saw an opportunity to simplify. Because file storage use cases are so closely correlated to data, we decided to embed [file attachments](https://xata.io/docs/concepts/file-attachments) directly into our serverless database and bring a new experience for building data apps that require binary storage.
+
+> Xata has become a critical part of my development cycle because I can quickly create and modify data structures without having to handle the database and migrations myself. Extending my records to include images has been great because I can easily apply transformations to them in the SDK. What I'm really excited about the ability to store any file type securely inside my database, it allows me to eliminate additional services like AWS S3.  
```

Comment:
Could we intro the quote with something like:

> We've shown this feature to a number of developers before releasing it, and this is the type of feedback we've received:

Reply:
Trying it out

Reply:
I'm not sure how I feel about this haha. What do you think?

Reply:
I like it like this, I think 3 is enough.